bento/crates/workflow/src/tasks/resolve_povw.rs

```diff
@@ -0,0 +1,175 @@
+// Copyright 2025 RISC Zero, Inc.
+//
+// Use of this source code is governed by the Business Source License
+// as found in the LICENSE-BSL file.
+
+use crate::{
+    Agent,
+    redis::{self, AsyncCommands},
+    tasks::{RECUR_RECEIPT_PATH, deserialize_obj, serialize_obj},
+};
+use anyhow::{Context, Result};
+use risc0_zkvm::sha::Digestible;
+use risc0_zkvm::{ReceiptClaim, SuccinctReceipt, Unknown, WorkClaim};
+use uuid::Uuid;
+use workflow_common::{KECCAK_RECEIPT_PATH, ResolveReq, s3::WORK_RECEIPTS_BUCKET_DIR};
+
+/// Run the POVW resolve operation
+pub async fn resolve_povw(
+    agent: &Agent,
+    job_id: &Uuid,
+    request: &ResolveReq,
+) -> Result<Option<u64>> {
+    let max_idx = &request.max_idx;
+    let job_prefix = format!("job:{job_id}");
+    let root_receipt_key = format!("{job_prefix}:{RECUR_RECEIPT_PATH}:{max_idx}");
+
+    tracing::debug!("Starting POVW resolve for job_id: {job_id}, max_idx: {max_idx}");
+
+    let mut conn = agent.redis_pool.get().await?;
+    let receipt: Vec<u8> = conn.get::<_, Vec<u8>>(&root_receipt_key).await.with_context(|| {
+        format!("segment data not found for root receipt key: {root_receipt_key}")
+    })?;
+
+    tracing::debug!("Root receipt size: {} bytes", receipt.len());
+
+    // Deserialize as POVW receipt
+    let povw_receipt: SuccinctReceipt<WorkClaim<ReceiptClaim>> =
+        deserialize_obj::<SuccinctReceipt<WorkClaim<ReceiptClaim>>>(&receipt)
+            .context("Failed to deserialize as POVW receipt")?;
+
+    // Unwrap the POVW receipt to get the ReceiptClaim for processing
+    let mut conditional_receipt: SuccinctReceipt<ReceiptClaim> =
+        if let Some(prover) = agent.prover.as_ref() {
+            prover.unwrap_povw(&povw_receipt).context(
+            "POVW unwrap method not available - POVW functionality requires RISC Zero POVW support",
```

Comment:
Is proving fails, this seems like it might be a confusing error messages, since their are other reasons this could fail. Also, because the prover is compiled into this binary I believe if PoVW is not available, the build will fail.